File: app/models/user_model.py
from pydantic import BaseModel, EmailStr, Field
from typing import Literal, Optional
from datetime import datetime

# ---- DB helper (MongoDB representation)
def user_helper(user) -> dict:
    return {
        "id": str(user["_id"]),
        "full_name": user["full_name"],
        "email": user["email"],
        "role": user.get("role", "candidate"),
        "created_at": user.get("created_at"),
    }


# ---- Pydantic models (for validation) ----

# Allowed roles
RoleType = Literal["admin", "recruiter", "candidate", "viewer"]

class UserBase(BaseModel):
    full_name: str
    email: EmailStr
    role: RoleType = "candidate"  # default role


class UserCreate(UserBase):
    password: str  # plain password (will be hashed)


class UserLogin(BaseModel):
    email: EmailStr
    password: str


class UserOut(UserBase):
    id: str
    created_at: datetime

    class Config:
        orm_mode = True


# ---- Optional: UserDB model (internal use, hashed password) ----
class UserDB(UserBase):
    id: str
    hashed_password: str
    created_at: datetime

    class Config:
        orm_mode = True

File: cv-management-frontend/src/app/models/user-model.ts
export interface User {
  id?: string;
  full_name: string;
  email: string;
  role?: string;        // optional: default 'candidate'
  password?: string;    // only sent to backend
  created_at?: string;
  token?: string;       // JWT token returned by backend
}

// --- Interfaces for Auth requests ---

export interface LoginRequest {
  email: string;
  password: string;
}

export interface RegisterRequest {
  full_name: string;
  email: string;
  password: string;
}
